extension-manifest-v3/src/utils/engines.js

```diff
@@ -406,107 +378,71 @@ async function update(name) {
     }
 
     if (updated) {
-      console.log(`Engine "${name}" updated`);
+      console.log(`[engines] Engine "${name}" updated`);
 
       // Notify listeners
-      const fns = updateListeners.get(name);
-      fns?.forEach((fn) => {
-        try {
-          fn();
-        } catch (e) {
-          console.error(`Error while calling update listener for "${name}"`, e);
-        }
-      });
+      notifyListeners(name);
 
       // Save the new engine to storage
       saveToStorage(name);
     }
 
     return engine;
   } catch (e) {
-    console.error(`Failed to update engine "${name}"`, e);
+    console.error(`[engines] Failed to update engine "${name}"`, e);
     throw e;
   }
 }
 
-export function updateAll() {
-  return Promise.all(Array.from(engines.keys()).map((name) => update(name)));
-}
-
 export function get(name) {
   return loadFromMemory(name);
 }
 
-const ALARM_PREFIX = 'engines:update:';
-const ALARM_DELAY = 60; // 1 hour
-
 export async function init(name) {
-  if (__PLATFORM__ === 'tests') return;
-
-  // Custom and regional engines are created on demand
-  // And should not be updated by alarms
-  if (name === CUSTOM_ENGINE || name === REGIONAL_ENGINE) {
-    return (
-      get(name) || (await loadFromStorage(name)) || (await createEngine(name))
-    );
-  }
-
-  // Schedule an alarm to update engines once an hour
-  chrome.alarms.get(`${ALARM_PREFIX}${name}`, (alarm) => {
-    if (!alarm) {
-      chrome.alarms.create(`${ALARM_PREFIX}${name}`, {
-        delayInMinutes: ALARM_DELAY,
-      });
-    }
-  });
-
   return (
-    get(name) || (await loadFromStorage(name)) || (await loadFromDisk(name))
+    get(name) ||
+    (await loadFromStorage(name)) ||
+    (name !== MAIN_ENGINE &&
+      name !== CUSTOM_ENGINE &&
+      (await loadFromDisk(name)))
```

Comment:
Perhaps we should rename "loadFromDisk" as loadBundledEngine (or something like that) to make it clear that it is the engine that comes as part of the release.

I find disk and storage confusing, because our storage is also on the local disk.

Reply:
I am not sure (but I am an author)... ;) What do you think about `loadFromFilesystem`?

Reply:
loadFromFile would also be clearer IMHO, yes.